detox/android/detox/src/main/java/com/wix/detox/espresso/DetoxAction.java

```diff
@@ -168,67 +169,24 @@ public void perform(UiController uiController, View view) {
         });
     }
 
-    private final static float EDGE_FUZZ_FACTOR = 0.083f;
-
     /**
      * Swipes the View in a direction.
      *
      * @param direction Direction to swipe (see {@link MotionDir})
      * @param fast true if fast, false if slow
-     *
+     * @param amount relative value from 0 to 1
+     * @param startOffsetPercentX relative value from 0 to 1
+     * @param startOffsetPercentY relative value from 0 to 1
      */
-    public static ViewAction swipeInDirection(final int direction, boolean fast) {
-        if (fast) {
-            switch (direction) {
-                case MOTION_DIR_LEFT:
-                    return swipeLeft();
-                case MOTION_DIR_RIGHT:
-                    return swipeRight();
-                case MOTION_DIR_UP:
-                    return swipeUp();
-                case MOTION_DIR_DOWN:
-                    return swipeDown();
-                default:
-                    throw new RuntimeException("Unsupported swipe direction: " + direction);
-            }
-        }
-
-        switch (direction) {
-            case MOTION_DIR_LEFT:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.CENTER_RIGHT, -EDGE_FUZZ_FACTOR, 0),
-                        GeneralLocation.CENTER_LEFT, Press.FINGER));
-            case MOTION_DIR_RIGHT:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.CENTER_LEFT, EDGE_FUZZ_FACTOR, 0),
-                        GeneralLocation.CENTER_RIGHT, Press.FINGER));
-            case MOTION_DIR_UP:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.BOTTOM_CENTER, 0, -EDGE_FUZZ_FACTOR),
-                        GeneralLocation.TOP_CENTER, Press.FINGER));
-            case MOTION_DIR_DOWN:
-                return actionWithAssertions(new GeneralSwipeAction(Swipe.SLOW,
-                        translate(GeneralLocation.TOP_CENTER, 0, EDGE_FUZZ_FACTOR),
-                        GeneralLocation.BOTTOM_CENTER, Press.FINGER));
-            default:
-                throw new RuntimeException("Unsupported swipe direction: " + direction);
-        }
+    public static ViewAction swipeInDirection(final int direction, boolean fast, double amount, double startOffsetPercentX, double startOffsetPercentY) {
+        final boolean isInCompatibilityMode = (fast && amount == Double.NaN && startOffsetPercentX == Double.NaN && startOffsetPercentY == Double.NaN);
+
+        return (isInCompatibilityMode)
+            ? SwipeHelper.INSTANCE.swipeFastInDirection(direction)
```

Comment:
I may be asking this simply because I don't understand this well enough just yet, but really - could we not give up on compatibility code and always use the new implementation, with proper fallbacks when offsets are not provided?

Reply:
Also, in a way, compatibility is needed here in case `fast=false`, isn't it?

Reply:
@d4vidi, well, `fast=true` triggers Espresso's built-in functions, while `fast=false` triggers some of our implementations. So, I felt safer about replacing our implementation rather than ditching the use of Espresso's facade functions while having no idea what's under the hood there.

Reply:
Okay, so, we decided to align iOS and Android behavior first, stepping on compatibility with previous Android implementation.

Reply:
Done.